db-tests/src/test/java/TestTriggers.java

```diff
@@ -252,12 +259,63 @@ void clearSimulations() throws SQLException {
     }
   }
 
+  int insertDataset() throws SQLException {
+    try (final var statement = connection.createStatement()) {
+      final var res = statement
+          .executeQuery(
+              """
+                  INSERT INTO dataset
+                  DEFAULT VALUES
+                  RETURNING id;"""
+          );
+      res.next();
+      return res.getInt("id");
+    }
+  }
+
+  void clearDatasets() throws SQLException {
+    try (final var statement = connection.createStatement()) {
+      statement
+          .executeUpdate(
+              """
+                  TRUNCATE dataset CASCADE;"""
+          );
+    }
+  }
+
+  Pair<Integer, Integer> insertPlanDataset(int planId) throws SQLException {
+    try (final var statement = connection.createStatement()) {
+      final var res = statement
+          .executeQuery(
+              """
+                  INSERT INTO plan_dataset (plan_id, offset_from_plan_start)
+                  VALUES ('%s', '0')
+                  RETURNING plan_id, dataset_id;"""
+                  .formatted(planId)
+          );
+      res.next();
+      return Pair.of(res.getInt("plan_id"), res.getInt("dataset_id"));
```

Comment:
Done

Reply:
Sorry to be so picky (though at the same times, we _should_ hold ourselves to high standards) but the name `PlanDatasetId` for the record seems misleading, as it's not the ID of a PlanDataset, but rather a record of a PlanDataset, which is at its most basic form just two IDs, the association of a plan ID with a Dataset ID. If you have opposing thoughts I'm happy to hear you out, but I do think we should rename it to `PlanDatasetRecord` (to be clear, not because it's a Java record, but because it's a literal record of data).